heron/spi/src/java/com/twitter/heron/spi/utils/NetworkUtils.java

```diff
@@ -31,23 +33,111 @@
 
 import com.twitter.heron.common.basics.Pair;
 import com.twitter.heron.common.basics.SysUtils;
+import com.twitter.heron.spi.common.Config;
 
 /**
  * Utilities related to network.
  */
 public final class NetworkUtils {
-  public static final String CONTENT_LENGTH = "Content-Length";
-  public static final String CONTENT_TYPE = "Content-Type";
-  public static final String LOCAL_HOST = "localhost";
+  private static final String CONTENT_LENGTH = "Content-Length";
+  private static final String CONTENT_TYPE = "Content-Type";
+  static final String LOCAL_HOST = "localhost";
 
   public static final String JSON_TYPE = "application/json";
   public static final String URL_ENCODE_TYPE = "application/x-www-form-urlencoded";
 
   private static final Logger LOG = Logger.getLogger(NetworkUtils.class.getName());
 
+  public enum TunnelType {
+    PORT_FORWARD,
+    SOCKS_PROXY,
+  }
+
+  public enum HeronSystem {
+    STATE_MANAGER("statemgr"),
+    SCHEDULER("scheduler");
+
+    private String shortName;
+
+    HeronSystem(String shortName) {
+      this.shortName = shortName;
+    }
+
+    public String getShortName() {
+      return shortName;
+    }
+  }
+
   private NetworkUtils() {
   }
 
+  public static class TunnelConfig {
```

Comment:
I could go either way on this. I made it an inner class since it's tightly couple to, and only intended for, `NetworkUtils` but if we think it should stand on it's own I'm ok with it.